### ISSUE TITLE
Error getting per-container exist status

### ISSUE THREAD

**Issue by abhik** [aws/amazon-ecs-agent#1021]
### Summary
In an ECS task with two containers, how can code running in one detect if the second container has stopped?

### Description
I'm running a task with two containers, `default` and `task`. `default` is essential and `task` is not. From within `default`, I would like to detect when `task` has exited. This appears possible with AWS APIs but the results are not as expected.

The AWS web console and awscli show both containers as running even though `docker container ls -a` on the instance shows one of the containers as exited. 

<img width="1409" alt="amazon_ec2_container_service" src="https://user-images.githubusercontent.com/11833/31697439-5c8324de-b36c-11e7-9787-a25e64cf3e4a.png">

```
[ec2-user@ip-10-0-2-234 /]$ docker container ls -a
CONTAINER ID        IMAGE                                       COMMAND                   CREATED             STATUS                           PORTS               NAMES
c0a4b9b54161        quay.io/lexentbio/ashah-sh:latest           "sh -c 'echo \"hell..."   48 minutes ago      Exited (0) 48 minutes ago                            ecs-ecs-launcher-test-13-task-d8f1f9c5e082db86f701
7797c55c758e        quay.io/lexentbio/ashah-test:latest         "sh -c /opt/entry.sh"     48 minutes ago      Up 48 minutes                                        ecs-ecs-launcher-test-13-default-c690abcbb3a6e9c28e01
```

```
$ aws ecs describe-tasks --cluster alignment-test --tasks fe5b4f30-30f8-4174-9f4e-11ac8fdcd5cf
{
    "failures": [],
    "tasks": [
        {
            "taskArn": "arn:aws:ecs:us-west-2:948305642098:task/fe5b4f30-30f8-4174-9f4e-11ac8fdcd5cf",
            "group": "family:ecs-launcher-test",
            "overrides": {
                "containerOverrides": [
                    {
                        "name": "task"
                    },
                    {
                        "name": "default"
                    }
                ]
            },
            "lastStatus": "RUNNING",
            "containerInstanceArn": "arn:aws:ecs:us-west-2:948305642098:container-instance/5cfa4e48-4f6f-4f23-ab81-c7fc5d01420d",
            "createdAt": 1508288424.144,
            "version": 2,
            "clusterArn": "arn:aws:ecs:us-west-2:948305642098:cluster/alignment-test",
            "startedAt": 1508288433.158,
            "desiredStatus": "RUNNING",
            "taskDefinitionArn": "arn:aws:ecs:us-west-2:948305642098:task-definition/ecs-launcher-test:13",
            "containers": [
                {
                    "containerArn": "arn:aws:ecs:us-west-2:948305642098:container/781a5d5b-1754-4b26-b5ca-56e9570abb0d",
                    "taskArn": "arn:aws:ecs:us-west-2:948305642098:task/fe5b4f30-30f8-4174-9f4e-11ac8fdcd5cf",
                    "lastStatus": "RUNNING",
                    "name": "task",
                    "networkBindings": []
                },
                {
                    "containerArn": "arn:aws:ecs:us-west-2:948305642098:container/b31af481-b625-4981-981d-e88cf8984d66",
                    "taskArn": "arn:aws:ecs:us-west-2:948305642098:task/fe5b4f30-30f8-4174-9f4e-11ac8fdcd5cf",
                    "lastStatus": "RUNNING",
                    "name": "default",
                    "networkBindings": []
                }
            ]
        }
    ]
}
```

### Versions

Running ECS Agent 1.14.5 and docker 17.03.2-ce.

**Comment by vsiddharth** [comment]
This issue should be resolved by #1026

**Comment by samuelkarp** [comment]
We have released [ECS agent v1.15.0](https://github.com/aws/amazon-ecs-agent/releases/tag/v1.15.0), which resolves this bug.
